src/snowflake/snowpark/dataframe.py

```diff
@@ -3496,6 +3497,41 @@ def with_column_renamed(self, existing: ColumnOrName, new: str) -> "DataFrame":
         ]
         return self.select(new_columns)
 
+    def qualify(self, expr: ColumnOrSqlExpr) -> "DataFrame":
```

Comment:
Just to brainstorm, is it possible to still use api `filter` and generate SQL with `qualify` when window functions are used?

Reply:
This is similar to Shixuan's question. To fully support that, we need to detect window expression in the current dataframe's projection, so for something like `df.select(window_expr.alias("winwin").filter(col("winwin") > 12)` we need to traverse the dataframe's logical plan. It's probably easier for simplified paths.